### ISSUE TITLE
Unable to "Sign In" through Xbox Application in Dev Mode

### ISSUE THREAD

**Issue by sulemanmuneerfraggames** [microsoft/xbox-live-unity-plugin#308]
Hi
I have been trying to use Xbox  Authentication in my project.
I have followed all the procedures provided in this link
https://docs.microsoft.com/en-us/gaming/xbox-live/get-started/setup-ide/creators/unity-win10/live-configure-xbl-in-unity
I added the "Title Configuration" manually because the association wizard wasn't working for me. But I successfully converted my system to developer mode.
I followed the procedure further and when I play the build on the Xbox dev mode and try to sign in, after selecting my account it throws an error.
![2A44DA30-75B8-4240-A924-1413E298AECA](https://user-images.githubusercontent.com/65340070/85831731-2a486e00-b7a8-11ea-89b6-1227ed39e3a4.jpg)

I have tried many things regarding this issue through internet but nothing worked for me so far.
If anyone knows about this issue kindly guide me.
Thanks.

**Comment by trdsolutions** [comment]
Thank god: Someone else has the same problem I deal with the whole day yesterday and still! 
I searched the whole internet for it and no solution came up to me. I was about to open an issue here as well. Sorry for not beeing helpful, but I hope you're as glad as I am, that you are not alone ;-)

**Comment by sulemanmuneerfraggames** [comment]
Yeah, it is relief that I am not alone. And maybe we could work it out together.
Can I ask you something?
Do you have the exact same issue and the same error on the Xbox Dev mode application while signing in?
Btw I followed the following links
https://forum.unity.com/threads/unable-to-sign-into-xbox-live-on-xbox-one.547693/
https://www.youtube.com/watch?v=ZtGWNPZuTtc
https://www.keepthetech.com/fix-problem-signing-in-to-the-xbox-app
and some other links but nothing worked for me. I also spent more than a day on this but could not figure it out. 
Most of the solutions on the internet are related to signing in issues in Xbox application on windows and their solutions are not relevant to this problem.

**Comment by trdsolutions** [comment]
I have the same error code, you will find below my stack trace.
I also read the first link, but I also added the `<uap:SupportedUsers>mutliple</uap:SupportedUsers>` and still have this problem.

I also identified some other error codes, where it says, that the app must associated with the Store App, which I did after every build, before remote debugging it on my Xbox One X.

I also watched the referenced YouTube video, but this isn't helpful for me neither. I'm already logged in at the Xbox and all settings are fine with my wired connection and NAT Type: Open. Just sign in in the game won't work :-(

Maybe you want to share your stack trace as well? So we have a better comparision

> onecore\base\appmodel\common\stateschema.cpp(64)\Windows.Storage.ApplicationData.dll!00007FFAF4D2DCE9: (caller: 00007FFAF4D1F3B9) LogHr(1) tid(a0c) 80070005 Access is denied.
"APPNAME.exe" (Win32): "C:\Windows\System32\AppXDeploymentClient.dll" geladen. 
onecoreuap\base\appmodel\statemanager\winrt\lib\windows.storage.applicationdatafactory.server.cpp(235)\Windows.Storage.ApplicationData.dll!00007FFAF4CF83D3: (caller: 00007FFAF4D1EEE1) ReturnHr(2) tid(a0c) 8000000B The operation attempted to access data outside the valid range
    Msg:[User S-1-5-21-2702878673-795188819-444038987-1003] 
"APPNAME.exe" (Win32): "C:\Windows\System32\XblAuthTokenBrokerExt.dll" geladen. 
"APPNAME.exe" (Win32): "C:\Windows\System32\XblAuthConsoleExt.dll" geladen. 
"APPNAME.exe" (Win32): "C:\Windows\System32\appmodeltools.dll" geladen. 
"APPNAME.exe" (Win32): "C:\Windows\System32\pkginstall.dll" geladen. 
"APPNAME.exe" (Win32): "C:\Windows\System32\xmllite.dll" geladen. 
"APPNAME.exe" (Win32): "C:\Windows\System32\srvcli.dll" geladen. 
"APPNAME.exe" (Win32): "C:\Windows\System32\XblAuthManagerProxy.dll" geladen. 
onecoreuap\xbox\xblauth\tokenbrokerext\dll\xblauth.cpp(824)\XblAuthTokenBrokerExt.dll!00007FFAFB6DC1C8: (caller: 00007FFAFB6DACC9) LogHr(1) tid(a0c) 80860003 The application requesting authentication tokens is either disabled or incorrectly configured.
onecoreuap\xbox\xblauth\tokenbrokerext\dll\xblauth.cpp(518)\XblAuthTokenBrokerExt.dll!00007FFAFB6DB36F: (caller: 00007FFAFB6DA543) ReturnHr(1) tid(a0c) 80070490 Element not found.
06/26/20 00:11:40 [1504] error  - Get token from IDP failed with ResponseStatus:3
[0310][I][16:00:14.666][HTTPCLIENT] Task queue completed: queueSize=1 taskId=1
[1480][I][16:00:14.667][HTTPCLIENT] HCTaskProcessNextCompletedTask: taskSubsystemId=128 taskGroupId=0 taskId=1
[1680][I][16:00:14.677][HTTPCLIENT] HCTaskCreate: taskGroupId=0 taskId=2
[1680][I][16:00:14.677][HTTPCLIENT] Task queue pending: queueSize=1 taskId=2
[0310][I][16:00:14.678][HTTPCLIENT] Task execute: executeQueueSize=1 taskId=2
"APPNAME.exe" (Win32): "C:\Windows\System32\Windows.StateRepositoryClient.dll" geladen. 
"APPNAME.exe" (Win32): "C:\Windows\System32\BCP47mrm.dll" geladen. 
onecore\base\appmodel\common\stateschema.cpp(64)\Windows.Storage.ApplicationData.dll!00007FFAF4D2DCE9: (caller: 00007FFAF4D1F3B9) LogHr(2) tid(a0c) 80070005 Access is denied.
onecoreuap\base\appmodel\statemanager\winrt\lib\windows.storage.applicationdatafactory.server.cpp(235)\Windows.Storage.ApplicationData.dll!00007FFAF4CF83D3: (caller: 00007FFAF4D1EEE1) ReturnHr(3) tid(a0c) 8000000B The operation attempted to access data outside the valid range
    Msg:[User S-1-5-21-2702878673-795188819-444038987-1003] 
onecoreuap\xbox\xblauth\tokenbrokerext\dll\xblauth.cpp(824)\XblAuthTokenBrokerExt.dll!00007FFAFB6DC1C8: (caller: 00007FFAFB6DACC9) LogHr(2) tid(a0c) 80860003 The application requesting authentication tokens is either disabled or incorrectly configured.
onecoreuap\xbox\xblauth\tokenbrokerext\dll\xblauth.cpp(518)\XblAuthTokenBrokerExt.dll!00007FFAFB6DB36F: (caller: 00007FFAFB6DA543) ReturnHr(2) tid(a0c) 80070490 Element not found.
[1.403025 / 22.079391] - OnWindowActivated event - Deactivated.
[0310][I][16:00:16.014][HTTPCLIENT] Task queue completed: queueSize=1 taskId=2
[1480][I][16:00:16.015][HTTPCLIENT] HCTaskProcessNextCompletedTask: taskSubsystemId=128 taskGroupId=0 taskId=2
Ausnahme ausgelöst bei 0x00007FFB10883DB9 in APPNAME.exe: Microsoft C++-Ausnahme: Il2CppExceptionWrapper bei Speicherort 0x000000D8621FDCF0.
XboxException: Xbox Services flat C API return error code -1000 with message "Provider error: Unexpected response code: Unknown, Error Code: 0x80131509"
> 
> Rethrow as AggregateException: One or more errors occurred.
  at Microsoft.Xbox.Services.Client.TaskYieldInstruction.get_keepWaiting () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.CustomYieldInstruction.MoveNext () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00000] in <00000000000000000000000000000000>:0 
>  
> (Filename: currently not available on il2cpp Line: -1)

**Comment by trdsolutions** [comment]
I also checked out the following link: https://docs.microsoft.com/en-us/gaming/xbox-live/features/identity/auth/live-troubleshooting-sign-in

There is mentioned, that the `80860003 The application requesting authentication tokens is either disabled or incorrectly configured` may be fixed through deleting the pfx-files (which I did) and double checking the App Store Association (which I also did many times!)

**Comment by trdsolutions** [comment]
@sulemanmuneerfraggames I made one step forward!
Try the following:

Double-click the package.appxmanifest file in the solution explorer and go to the last tab (something like package or so, I'm german) and there you click "Choose certificate..." (or similar). On the opened window you choose "Create..." and as publisher name you enter your publisher key from the Dev App Center (https://partner.microsoft.com/de-de/dashboard/account/management) (it's the windows publisher id without "CN=").
Then rebuild the application.
Then associate it with the store again and test it again.

Did it help?

**Comment by sulemanmuneerfraggames** [comment]
> 
> 
> @sulemanmuneerfraggames I made one step forward!
> Try the following:
> 
> Double-click the package.appxmanifest file in the solution explorer and go to the last tab (something like package or so, I'm german) and there you click "Choose certificate..." (or similar). On the opened window you choose "Create..." and as publisher name you enter your publisher key from the Dev App Center (https://partner.microsoft.com/de-de/dashboard/account/management) (it's the windows publisher id without "CN=").
> Then rebuild the application.
> Then associate it with the store again and test it again.
> 
> Did it help?

Let me try this one.

**Comment by sulemanmuneerfraggames** [comment]
> 
> 
> > @sulemanmuneerfraggames I made one step forward!
> > Try the following:
> > Double-click the package.appxmanifest file in the solution explorer and go to the last tab (something like package or so, I'm german) and there you click "Choose certificate..." (or similar). On the opened window you choose "Create..." and as publisher name you enter your publisher key from the Dev App Center (https://partner.microsoft.com/de-de/dashboard/account/management) (it's the windows publisher id without "CN=").
> > Then rebuild the application.
> > Then associate it with the store again and test it again.
> > Did it help?
> 
> Let me try this one.



> 
> 
> > @sulemanmuneerfraggames I made one step forward!
> > Try the following:
> > Double-click the package.appxmanifest file in the solution explorer and go to the last tab (something like package or so, I'm german) and there you click "Choose certificate..." (or similar). On the opened window you choose "Create..." and as publisher name you enter your publisher key from the Dev App Center (https://partner.microsoft.com/de-de/dashboard/account/management) (it's the windows publisher id without "CN=").
> > Then rebuild the application.
> > Then associate it with the store again and test it again.
> > Did it help?
> 
> Let me try this one.

**Comment by sulemanmuneerfraggames** [comment]
I accidentally closed this issue, sorry about that.
I have already created a certificate for my game. But I guess I have not provided this publisher key you are referring to. Should I create a new one?

**Comment by trdsolutions** [comment]
> I accidentally closed this issue, sorry about that.
> I have already created a certificate for my game. But I guess I have not provided this publisher key you are referring to. Should I create a new one?

Yes, just create a new one with the publisher key

**Comment by sulemanmuneerfraggames** [comment]
Still not working for me.

**Comment by trdsolutions** [comment]
Okay, what about your stack trace? Start your game on the Xbox console via Remote-Debugging from within Visual Studio and if look at your output in Visual Studio when you try to login. Copy all of it from the point where you triggered the login function to the -1000 error code and paste it in here and we will have a look toghether at it

**Comment by trdsolutions** [comment]
Did you also include `"XboxLiveCreatorsTitle": true` in your XboxServices.config? I missed that as well yesterday, because I also needed to fill out that file manually due still not working AssociationWizard.

**Comment by sulemanmuneerfraggames** [comment]
> 
> 
> Did you also include `"XboxLiveCreatorsTitle": true` in your XboxServices.config? I missed that as well yesterday, because I also needed to fill out that file manually due still not working AssociationWizard.

Yeah I have included that. I missed that before.

**Comment by sulemanmuneerfraggames** [comment]
> 
> 
> Okay, what about your stack trace? Start your game on the Xbox console via Remote-Debugging from within Visual Studio and look at your output in Visual Studio when you try to login. Copy all of it from the point where you triggered the login function to the -1000 error code and paste it in here and we will have a look toghether at it

Okay I'll try that next.

**Comment by trdsolutions** [comment]
After modifying the `XboxService.config` and or the `Package.appxmanifest` run the App Store association again; I had the impression, that it changes something.

**Comment by ImperialDynamicsOfficial** [comment]
yes the association wizard is useless, one has to edit the xboxservice.config file manually.
And generally this plugin is in serious need of a long overdue update.

**Comment by fuzzymannerz** [comment]
I'm getting the same issue. Been trying various solutions for a few days now. **Very** frustrating.
`XboxException: Xbox Services flat C API return error code -1000 with message "Provider error: Unexpected response code: Unknown, Error Code: 0x80131509"`

Using Unity _2020.1.2f1_ and the fixed plugin from https://github.com/ChossyUK/Unity-XBL-Plugin-Fixed.

My `XboxServices.config` as far as I know seems correct, as does the information and additional line in the properties section (`<uap:SupportedUsers>multiple</uap:SupportedUsers>`) in `Package.appx.manifest`. I have also tried the "Associate App with the Store" from VS.

**Comment by layers764** [comment]
> m

Ive been able to manually configure it all except for one thing
how do I get the
"AppId" field?

**Comment by MSFT-Heba** [comment]
Hey everyone,
Just for awareness, we've posted an announcement that integrating Xbox Live is no longer a requirement for UWP games to be published on Xbox as part of the Creators Program and we are phasing out software needed to support Xbox Live on UWP. 

For more details: https://github.com/microsoft/xbox-live-unity-plugin#2021-update-more-accessible-creators-program-no-longer-requires-xbox-live

**Comment by hangshow2011** [comment]
> Hey everyone, Just for awareness, we've posted an announcement that integrating Xbox Live is no longer a requirement for UWP games to be published on Xbox as part of the Creators Program and we are phasing out software needed to support Xbox Live on UWP.
> 
> For more details: https://github.com/microsoft/xbox-live-unity-plugin#2021-update-more-accessible-creators-program-no-longer-requires-xbox-live

Does GDk support unity3d?
